### PR TITLE
PRfD - Create PR page

### LINKED ISSUE #8174: PRfD - WebUI - Create PR page
Create page for creating a new PR (which will be opened by the button in the PRs list page).

This includes:
TBD.

### PR DESCRIPTION
**Pull request by itaigilo**
Closes #8174.

## Change Description

Adding the page for creating a Pull Request.

-----

Screenshot 2024-09-22 at 16 08 04

### CODE REVIEW

File: webui/src/lib/api/index.js
Comment (itaigilo) [inline webui/src/lib/api/index.js:572]: Yeah already integrated the other APIs as part of this one.

File: webui/src/lib/components/repository/compareBranchesActionBar.jsx
Comment (itaigilo) [inline webui/src/lib/components/repository/compareBranchesActionBar.jsx:19]: The `Compare` and `Create PR` pages are both using the same component of branches selection,
Only with different baseURL and slightly different Ref params.
So I've unified them.

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (itaigilo) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:43]: As part of the integration, update the date processing.

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Nice refactoring!

In addition to the comment above please address the following before we could merge this:

**Pull request creation:**
Pull contents (title and description) deleted when changing branches
You can open a pull request even if there are no changes or it's the same branch (button should be greyed out)

**Pull request view:**
Can we align the pull request description with the changes view (same width)
Lets not show pull ID in the GUI (it's not aesthetic)
We have a bad error when deleting a PR branch (we should print an informative message there and grey out the Merge button
image

**Comment by itaigilo** [comment]
> Pull contents (title and description) deleted when changing branches
Since the url changes in this case, not sure how easy it is to implement this (I think it needs to be saved in the localStorage or something). I honestly don't think it's worth the effort for the MVP.

> You can open a pull request even if there are no changes or it's the same branch (button should be greyed out)
True. I'll handle it.

> Can we align the pull request description with the changes view (same width)
This might make sense, thanks. I'll play around with it to see how it looks.

> Lets not show pull ID in the GUI (it's not aesthetic)
Yeah I was also wondering about it. If you also vote for removing it, I will.

> We have a bad error when deleting a PR branch (we should print an informative message there and grey out the Merge button
Regarding the error message - do you think we should print a generic error instead? Something like "Pull Request creation failed"? O can you think of a nicer way of presenting server errors?
And as for disabling the Merge button after error - I don't think we should. A user should be able to retry, plus they can just refresh the page and retry. So I think it should remain enabled.

File: webui/src/lib/api/index.js
Comment (itaigilo) [reply webui/src/lib/api/index.js:584]: And wrap it up with an error message for the user here?

File: webui/src/lib/api/index.js
Comment (itaigilo) [reply webui/src/lib/api/index.js:598]: Took me a while to figure out how to properly get the id in JS 😖 

No problem, update me when your change is in.

**Comment by N-o-Z** [comment]
> > Pull contents (title and description) deleted when changing branches
> 
> Since the url changes in this case, not sure how easy it is to implement this (I think it needs to be saved in the localStorage or something). I honestly don't think it's worth the effort for the MVP.
> 
> > You can open a pull request even if there are no changes or it's the same branch (button should be greyed out)
> 
> True. I'll handle it.
> 
> > Can we align the pull request description with the changes view (same width)
> 
> This might make sense, thanks. I'll play around with it to see how it looks.
> 
> > Lets not show pull ID in the GUI (it's not aesthetic)
> 
> Yeah I was also wondering about it. If you also vote for removing it, I will.
> 
> > We have a bad error when deleting a PR branch (we should print an informative message there and grey out the Merge button
> 
> Regarding the error message - do you think we should print a generic error instead? Something like "Pull Request creation failed"? O can you think of a nicer way of presenting server errors? And as for disabling the Merge button after error - I don't think we should. A user should be able to retry, plus they can just refresh the page and retry. So I think it should remain enabled.

Thanks,

Regarding the last comment.
What I meant was, after opening the pull request if one of the branches are deleted we get an ugly error in the changes view. We should optimally just provide a descriptive message saying one of the branches was deleted and disable the Merge button on that case. The same goes if there are no changes between the branches - allowing the user to try and merge when there are no changes is just not good user experience.
I'm willing to defer the latter if it's too complicated for now but let's open an issue on it if so

File: webui/src/lib/api/index.js
Comment (N-o-Z) [reply webui/src/lib/api/index.js:584]: Oh don't worry you'll know since you'll be the one approving 😜

File: webui/src/lib/api/index.js
Comment (N-o-Z) [reply webui/src/lib/api/index.js:598]: Oh don't worry you'll know since you'll be the one approving 😜

File: webui/src/lib/api/index.js
Comment (N-o-Z) [reply webui/src/lib/api/index.js:584]: Just like in `get`, throw the explicit Error type for that status

File: webui/src/lib/api/index.js
Comment (N-o-Z) [inline webui/src/lib/api/index.js:585]: I don't think it's a good idea to write to console

File: webui/src/lib/api/index.js
Comment (N-o-Z) [inline webui/src/lib/api/index.js:597]: Same here

File: webui/src/lib/api/index.js
Comment (N-o-Z) [inline webui/src/lib/api/index.js:595]: Check for error codes and provide the message in the resp. This is very confusing:
image
The server does return the reason for 400

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (N-o-Z) [inline webui/src/pages/repositories/repository/pulls/pullDetails.jsx:41]: Still need to fix the changes view when branch is deleted:
image

**Review by N-o-Z — APPROVED** [review]
Approved to unblock but please make sure to fix all comments before merging!

File: webui/src/pages/repositories/repository/pulls/pullDetails.jsx
Comment (itaigilo) [reply webui/src/pages/repositories/repository/pulls/pullDetails.jsx:41]: Will be handled when handling empty diffs. Same area.

**Comment by itaigilo** [comment]
All handled,
Except for the comments related to empty diffs, which will be handled in a separate PR.

**Pull request merged**
